bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java

```diff
@@ -732,74 +319,72 @@ private String quantityTypeToDPTValue(QuantityType<?> qt, int mainNumber, int su
     }
 
-    @Override
-    public @Nullable String toDPTValue(Type type, @Nullable String dptID) {
+    /**
+     * formats the given value as String for outputting via Calimero
+     *
+     * @param value the value
+     * @param dptId the DPT id to use for formatting the string (e.g. 9.001)
+     * @return the value formatted as String
+     */
+    public static @Nullable String formatAsDPTString(Type value, String dptId) {
         DPT dpt;
-        int mainNumber = getMainNumber(dptID);
-        if (mainNumber == -1) {
-            logger.error("toDPTValue couldn't identify mainnumber in dptID: {}", dptID);
-            return null;
-        }
-        int subNumber = getSubNumber(dptID);
-        if (subNumber == -1) {
-            logger.debug("toType: couldn't identify sub number in dptID: {}.", dptID);
+
+        Matcher m = DPT_PATTERN.matcher(dptId);
+        if (!m.matches() || m.groupCount() != 2) {
+            LOGGER.warn("formatAsDPTString couldn't identify main/sub number in dptId '{}'", dptId);
             return null;
         }
 
+        String mainNumber = m.group("main");
+
         try {
-            DPTXlator translator = TranslatorTypes.createTranslator(mainNumber, dptID);
+            DPTXlator translator = TranslatorTypes.createTranslator(Integer.parseInt(mainNumber), dptId);
             dpt = translator.getType();
         } catch (KNXException e) {
             return null;
         }
 
         try {
             // check for HSBType first, because it extends PercentType as well
-            if (type instanceof HSBType) {
-                switch (mainNumber) {
-                    case 5:
-                        switch (subNumber) {
-                            case 3: // * 5.003: Angle, values: 0...360 °
-                                return ((HSBType) type).getHue().toString();
-                            case 1: // * 5.001: Scaling, values: 0...100 %
-                            default:
-                                return ((HSBType) type).getBrightness().toString();
-                        }
-                    case 232:
-                        switch (subNumber) {
-                            case 600: // 232.600
-                                HSBType hc = ((HSBType) type);
-                                return "r:" + convertPercentToByte(hc.getRed()) + " g:"
-                                        + convertPercentToByte(hc.getGreen()) + " b:"
-                                        + convertPercentToByte(hc.getBlue());
-                        }
-                    default:
-                        HSBType hc = ((HSBType) type);
-                        return "r:" + hc.getRed().intValue() + " g:" + hc.getGreen().intValue() + " b:"
-                                + hc.getBlue().intValue();
+            if (value instanceof HSBType) {
+                HSBType hsb = ((HSBType) value);
```

Comment:
Use pattern matching:

```suggestion
            if (value instanceof HSBType hsb) {
```

Reply:
thanks @J-N-K. that's a nice java17 feature... I have now used this also for the other occurrences.